mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReprocessingContextInformationDTO.java

```diff
@@ -199,6 +205,13 @@ static ReIndexingExecutionFailures deserializeFailures(MailboxId.Factory mailbox
             .collect(Guavate.toImmutableList());
     }
 
+    static List<ReindexingFailureDTO> resolveFailure(Optional<List<ReindexingFailureDTO>> failures, Optional<List<ReindexingFailureDTO>> messageFailures) {
+        Preconditions.checkState(Booleans.countTrue(failures.isPresent(), messageFailures.isPresent()) == 1,
```

Comment:
or you can use the XOR operator.
`failures.isPresent() ^ messageFailures.isPresent()`
But no need to change that.